addons/base_automation/models/base_automation.py

```diff
@@ -1006,22 +1033,61 @@ def get_record_dt(record):
                             last_run,
                             compute_leaves=True,
                         )
-                    is_process_to_run = past_last_run[calendar.id] <= fields.Datetime.to_datetime(record_dt) < past_now[calendar.id]
+                    return past_last_run[calendar.id] <= record_dt < past_now[calendar.id]
+
+                records = records.filtered_domain(calendar_filter)
```

Comment:
Whut ? How does it word, filtered_domain only accept domain :thinking: 
`filtered` do you mean ? Maybe a good time to test this case ?